protocols/ping/src/protocol.rs

```diff
@@ -63,10 +65,10 @@ where
 
     fn upgrade_inbound(self, mut socket: TSocket, _: Self::Info) -> Self::Future {
         async move {
-            let mut payload = [0u8; 32];
-            socket.read_exact(&mut payload).await?;
-            socket.write_all(&payload).await?;
-            socket.close().await?;
+            let mut payload = [0u8; PING_SIZE];
+            while let Ok(_) = socket.read_exact(&mut payload).await {
```

Comment:
Will this not return an unexpected EOF error when the remote closes the stream after a series of pings?

Reply:
It swallows all errors and just returns Ok(()) as soon as read_exact returns anything other than Ok. In my experiments with go-ipfs it was not a problem.

Reply:
Ah yeah, sorry. Should have read more carefully.